.github/workflows/ci.yml

```diff
@@ -1,8 +1,16 @@
-# This file was generated on 2020-11-16T09:31:10+00:00 from the rspec-dev repo.
+# This file was generated on 2020-11-16T21:55:30+00:00 from the rspec-dev repo.
 # DO NOT modify it by hand as your changes will get lost the next time it is generated.
 
 name: RSpec CI
-on: [pull_request, push]
+on:
+  push:
+    branches:
+      - 'main'
+      - '*-maintenance'
+      - '*-dev'
+  pull_request:
+    branches:
+      - '*'
```

Comment:
@pirj @benoittgt this seems to work for pull requests at least with just one build, will need to push to a branch (probably maintenance to cherry-pick the config) to test it still builds pushes.